src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/TestVectors.java

```diff
@@ -0,0 +1,216 @@
+package com.sap.sgs.phosphor.fosstars.model.qa;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A collection of test vectors.
+ */
+public class TestVectors implements Iterable<TestVector> {
+
+  /**
+   * No default values.
+   */
+  private static final Set<Value> NO_DEFAULTS = Collections.emptySet();
+
+  /**
+   * A factory for parsing YAML.
+   */
+  private static final YAMLFactory YAML_FACTORY;
+
+  /**
+   * For serialization and deserialization in YAML.
+   */
+  static final ObjectMapper YAML_OBJECT_MAPPER;
+
+  static {
+    YAML_FACTORY = new YAMLFactory();
+    YAML_FACTORY.disable(YAMLGenerator.Feature.USE_NATIVE_TYPE_ID);
+    YAML_OBJECT_MAPPER = new ObjectMapper(YAML_FACTORY);
+    YAML_OBJECT_MAPPER.findAndRegisterModules();
+  }
+
+  /**
+   * A list of test vectors.
+   */
+  private final List<TestVector> vectors;
+
+  /**
+   * A set of default values.
+   */
+  private final Set<Value> defaults;
+
+  /**
+   * Initializes a new collection of test vectors.
+   *
+   * @param vectors The test vectors.
+   */
+  public TestVectors(TestVector... vectors) {
+    this(Arrays.asList(vectors), NO_DEFAULTS);
+  }
+
+  /**
+   * Initializes a collection of test vectors.
+   *
+   * @param vectors The test vectors.
+   * @param defaults A set of default values.
+   */
+  public TestVectors(
+      @JsonProperty("vectors") List<TestVector> vectors,
+      @JsonProperty(value = "defaults") Set<Value> defaults) {
+
+    Objects.requireNonNull(vectors, "Hey! Vectors can't be null!");
+    Objects.requireNonNull(defaults, "Hey! Defaults can't be null!");
+    this.vectors = new ArrayList<>(vectors);
+    this.defaults = new HashSet<>(defaults);
+  }
+
+  /**
+   * Adds a test vector to the collection.
+   *
+   * @param vector The test vector to be added.
+   */
+  public void add(TestVector vector) {
+    vectors.add(vector);
+  }
+
+  /**
+   * Adds test vectors to the collection.
+   *
+   * @param vectors The test vectors to be added.
+   */
+  public void add(TestVectors vectors) {
+    for (TestVector vector : vectors.vectors) {
```

Comment:
A personal opinion, `vectors.vectors` seems a bit weird.
You can ignore this or maybe change the name of the variable?

Reply:
You are right. I'll update it to `elements`.